ui-components/src/helpers/tableSummaries.tsx

```diff
@@ -1,81 +1,145 @@
-      const unit = percent ? `% ${t("t.income")}` : ` ${t("t.perMonth")}`
-      return rentMin == rentMax ? (
-        <>
-          <strong>{rentMin}</strong>
-          {unit}
-        </>
-      ) : (
-        <>
-          <strong>{rentMin}</strong> {t("t.to")} <strong>{rentMax}</strong>
-          {unit}
-        </>
-      )
+  const getRent = (rentMin?: string, rentMax?: string, percent = false) => {
+    const unit = percent ? `% ${t("t.income")}` : ` ${t("t.perMonth")}`
+    if (rentMin == undefined && rentMax == undefined) {
+      return <strong>Data Not Here</strong>
     }
+    return rentMin == rentMax ? (
+      <>
+        <strong>{rentMin}</strong>
+        {unit}
+      </>
+    ) : (
+      <>
+        <strong>{rentMin}</strong> {t("t.to")} <strong>{rentMax}</strong>
+        {unit}
+      </>
+    )
+  }
 
-    // Use rent as percent income if available, otherwise use exact rent
-    const rent = unitSummary.rentAsPercentIncomeRange.min
-      ? getRent(
-          unitSummary.rentAsPercentIncomeRange.min.toString(),
-          unitSummary.rentAsPercentIncomeRange.max.toString(),
-          true
-        )
-      : getRent(unitSummary.rentRange.min, unitSummary.rentRange.max)
+  // Use rent as percent income if available, otherwise use exact rent
+  const rent = rentAsPercentIncomeRangeMin
+    ? getRent(rentAsPercentIncomeRangeMin, rentAsPercentIncomeRangeMax, true)
+    : getRent(rentRangeMin, rentRangeMax)
 
-    return {
-      unitType: <strong>{t(`listings.unitTypes.${unitSummary.unitType.name}`)}</strong>,
-      minimumIncome: (
-        <>
-          {minIncome} {t("t.perMonth")}
-        </>
-      ),
-      rent: <>{rent}</>,
-      availability: (
-        <>
-          {unitSummary.totalAvailable > 0 ? (
-            <>
-              <strong>{unitSummary.totalAvailable}</strong>{" "}
-              {unitSummary.totalAvailable == 1 ? t("t.unit") : t("t.units")}
-            </>
-          ) : (
-            <span className="uppercase">{t("listings.waitlist.label")}</span>
-          )}
-        </>
-      ),
-      totalCount: (
-        <>
-          <strong>{unitSummary.totalCount}</strong>{" "}
-          {unitSummary.totalCount == 1 ? t("t.unit") : t("t.units")}
-        </>
-      ),
-    }
+  return {
+    unitType: <strong>{t(`listings.unitTypes.${unitTypeName}`)}</strong>,
+    minimumIncome: <>{minIncome}</>,
+    rent: <>{rent}</>,
+    availability: (
+      <>
+        {totalAvailable && totalAvailable > 0 ? (
+          <>
+            <strong>{totalAvailable}</strong> {totalAvailable == 1 ? t("t.unit") : t("t.units")}
+          </>
+        ) : (
+          <span className="uppercase">{t("listings.waitlist.label")}</span>
+        )}
+      </>
+    ),
+    totalCount: (
+      <>
+        <strong>{totalCount}</strong> {totalCount == 1 ? t("t.unit") : t("t.units")}
+      </>
+    ),
+  }
+}
+
+export const unitSummariesTableFromUnitSummary = (summaries: UnitSummary[]) => {
```

Comment:
Can we get rid of this function by just putting the logic for this into `getSummariesTableFromUnitSummary`?

Reply:
Updated.